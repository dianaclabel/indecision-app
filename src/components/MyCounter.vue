<template>
  <seccion>
    <h3>Counter : {{ counter }}</h3>
    <h3 data-testid="square-label">Square : {{ squareCounter }}</h3>
    <div>
      <button class="btn" @click="counter++">+1</button>
      <button data-testid="btnDecrement" class="btn" @click="counter--">-1</button>
    </div>
  </seccion>
</template>

<script lang="ts" setup>
//utiliza composable
import { useCounter } from '@/composables/useCounter'

interface Props {
  value: number
}

const props = defineProps<Props>()

const { counter, squareCounter } = useCounter(props.value)
</script>
<style scoped>
.btn {
  @apply p-5 bg-red-500 rounded hover:bg-red-700 mr-2;
}
</style>
