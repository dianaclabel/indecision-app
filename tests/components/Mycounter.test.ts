import { test, describe } from 'vitest'
import { mount } from '@vue/test-utils'
import MyCounter from '../../src/components/MyCounter.vue'

describe('<My counter/> ', () => {
  test('should match snapshot', () => {
    const wrapper = mount(MyCounter, {
      props: {
        value: 10
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('renders the counter value correctly', () => {
    const value = 10
    const wrapper = mount(MyCounter, {
      props: {
        value: value
      }
    })

    expect(wrapper.find('h3').text()).toContain(`Counter : ${value}`)
    expect(wrapper.find('[data-testid="square-label"]').text()).toContain(
      `Square : ${value * value}`
    )

    const [counterLabel, squareLabel] = wrapper.findAll('h3')

    expect(counterLabel.text()).toContain(`Counter : ${value}`)
    expect(squareLabel.text()).toContain(`Square : ${value * value}`)
  })

  test('increments the counter when +1 button is clicked', async () => {
    const value = 10
    const wrapper = mount(MyCounter, {
      props: {
        value: value
      }
    })

    const [counterLabel, squareLabel] = wrapper.findAll('h3')

    const btnIncrement = wrapper.find('button')
    await btnIncrement.trigger('click')

    expect(counterLabel.text()).toContain(`Counter : ${value + 1}`)
    expect(squareLabel.text()).toContain(`Square : ${(value + 1) * (value + 1)}`)
  })

  test('decrements the counter when -1 button is clicked twice', async () => {
    const value = 10
    const wrapper = mount(MyCounter, {
      props: {
        value: value
      }
    })

    const [counterLabel, squareLabel] = wrapper.findAll('h3')

    const [, btnDecrement] = wrapper.findAll('button')

    await btnDecrement.trigger('click')
    await btnDecrement.trigger('click')

    expect(counterLabel.text()).toContain(`Counter : ${value - 2}`)
    expect(squareLabel.text()).toContain(`Square : ${(value - 2) * (value - 2)}`)
  })
})
