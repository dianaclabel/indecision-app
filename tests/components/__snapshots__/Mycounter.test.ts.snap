// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<My counter/>  > should match snapshot 1`] = `
"<seccion data-v-0352012b="">
  <h3 data-v-0352012b="">Counter : 10</h3>
  <h3 data-v-0352012b="" data-testid="square-label">Square : 100</h3>
  <div data-v-0352012b=""><button data-v-0352012b="" class="btn">+1</button><button data-v-0352012b="" data-testid="btnDecrement" class="btn">-1</button></div>
</seccion>"
`;
